earthkit/transforms/tools.py

```diff
@@ -65,7 +66,17 @@ def wrapper(
 
             dataarray = dataarray.assign_coords({time_dim: time_coord})
 
-        return func(dataarray, *args, time_dim=time_dim, **kwargs)
+        result = func(dataarray, *args, time_dim=time_dim, **kwargs)
+
+        # If we want only full days then chop off the first and last day if we have time_shift
+        if (
+            remove_partial_periods
+            and time_shift is not None
+            and isinstance(result, (xr.Dataset, xr.DataArray))
+        ):
+            return result.isel({time_dim: slice(1, -1)})
```

Comment:
Is this safe? What if the time shift is 24 hours? Should we double-check that each end chunk is definitely not a complete time chunk?

Reply:
Well this is why I left it as a switch, I'm not sure we could make a safe detector, the time is not necessarily contiguous, and I don't want to make that assumption